File: 2231-find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.java
class Solution {
    public String firstPalindrome(String[] words) {

        for (String ele : words) {
            int start = 0;
            int end = ele.length() - 1;
            boolean flag = true;

            while (start < end) {
                if (ele.charAt(start) != ele.charAt(end)) {
                    flag = false;
                    break;
                }
                start++;
                end--;
            }

            if (flag)
                return ele;
        }
        return "";
    }
}

File: 2231-find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.py
class Solution(object):
    def firstPalindrome(self, words):
        
        for ele in words:
            start = 0
            end = len(ele)-1
            flag = True

            while start < end:
                if ele[start] != ele[end]:
                    flag = False
                    break
                start += 1
                end -= 1
            
            if flag:
                return ele
        
        return ""
        